workers/common/src/main/kotlin/common/context/WorkerContextImpl.kt

```diff
@@ -134,10 +140,22 @@ internal class WorkerContextImpl(
         config: Map<String, ResolvablePluginConfig>?
     ): Map<String, PluginConfig> =
         config?.let { c ->
-            val secrets = c.values.flatMap { pluginConfig -> pluginConfig.secrets.values.map { it.name } }
-            val resolvedSecrets = parallelTransform(secrets, configSecretsCache, this::resolveConfigSecret) { it }
+            val configSecrets = c.values.flatMap { pluginConfig ->
```

Comment:
Maybe use `partition` here to distinguish between config and user secrets (assuming that there are only two sources)?